### PR TITLE
chore: move eslint to managedDeps

### PR DESCRIPTION
**Pull request by viliusl**
let's wait for ci - it might just fail:)

### CODE REVIEW

File: plugins/yoshi-eslint/package.json [hunk 1]
```diff
@@ -12,12 +12,12 @@
     "registry": "https://registry.npmjs.org/"
   },
   "dependencies": {
-    "eslint": "^3.16.1",
+    "eslint": "~4.2.0",
```
Comment (roymiloh) [inline plugins/yoshi-eslint/package.json]: @viliusl Are we sure that this change won't break yoshi-eslint's clients?

**Comment by viliusl** [comment]
@roymiloh no. I'm just checking feasibility of it and even our modules break. Q is - do we stay on old version forever, or maybe just run it on couple of bigger projects?

**Comment by roymiloh** [comment]
I usually test these kind of things on several big projects, but the question is still relevant, so we need to decide how we roll breaking changes.
I'd usually go with ft, but this case (upgrading major version) is a bit tricky, we can hack things around with `postinstall`, so if the ft is enabled install the new one, otherwise the old one and use the API accordingly (if there are any changes in this aspect).

**Comment by viliusl** [comment]
@roymiloh - I decided for now just to unify and not break things as there are quite some breaking changes and linting will fail for many. This will have to handled eventually, but afaik it is out of scope for now.

@ronami please

**Pull request merged**